images/router/haproxy/conf/haproxy-config.template

```diff
@@ -184,6 +184,22 @@ defaults
   option contstats
   {{- end }}
 
+  # Prevent failures at startup due to name resolution
+  default-server init-addr last,none resolvers ingress_dns
+
+# Define a resolver to be used when servers are specified by name and
+# therefore requires name resolution. Server name resolution typically
+# occurs only at startup. With this configuration, name resolution
+# will occur once every 10s. Successful name resolution results in the
+# server being marked UP while failure results in the server being
+# marked DOWN. Note this only occurs when servers are specified by
+# name.
+resolvers ingress_dns
+  parse-resolv-conf
+  resolve_retries 1
```

Comment:
The documentation I've found on the `resolvers` block is a bit sparse, but my reading didn't seem to suggest that haproxy would continue to try resolving dns names after running out of retries.

I assume you've tested this, but can you confirm that you do see haproxy continuing to try to resolve servers after an initial failure?